sql/catalyst/src/main/scala/org/apache/spark/sql/internal/StaticSQLConf.scala

```diff
@@ -132,6 +132,19 @@ object StaticSQLConf {
       .intConf
       .createWithDefault(1000)
 
+  val BROADCAST_EXCHANGE_MAX_THREAD_THREASHOLD =
+    buildStaticConf("spark.sql.broadcastExchange.maxThreadThreshold")
+      .internal()
+      .doc("The maximum degree of parallelism to fetch and broadcast the table. " +
+        "If we encounter memory issue like frequently full GC or OOM when broadcast table " +
+        "we can decrease this number in order to reduce memory usage. " +
+        "Notice the number should be carefully chosen since decreasing parallelism might " +
+        "cause longer waiting for other broadcasting. Also, increasing parallelism may " +
+        "cause memory problem.")
+      .intConf
+      .checkValue(thres => thres > 0 && thres <= 128, "The threshold should be positive.")
```

Comment:
There is a maximum limit (128) for this config.  The error message doesn't reflect that. Once an invalid value like 200 is set, it can't figure out what goes wrong with it.

Reply:
Yea let's improve this one before getting this in

Reply:
Done @HyukjinKwon @viirya Thanks!